src/Acuminator/Acuminator.Utils/RoslynExtensions/PXFieldAttributes/AttributeInformation.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Acuminator.Analyzers;
+using Microsoft.CodeAnalysis;
+
+namespace Acuminator.Utilities
+{
+	/// <summary>
+	/// Information about the Acumatica field attributes.
+	/// </summary>
+	public class AttributeInformation
+	{
+		private readonly PXContext _context;
+
+
+		public AttributeInformation(PXContext pxContext)
+		{
+			pxContext.ThrowOnNull(nameof(pxContext));
+
+			_context = pxContext;
+		}
+
+		public bool ContainsBaseType(ITypeSymbol attributeSymbol, ITypeSymbol type)
+		{
+			attributeSymbol.ThrowOnNull(nameof(attributeSymbol));
+			type.ThrowOnNull(nameof(type));
+
+			List<ITypeSymbol> attributeTypeHierarchy = attributeSymbol.GetBaseTypesAndThis().ToList();
+			if (attributeTypeHierarchy.Contains(type))
+				return true;
+			return false;
+		}
+
+		public bool AttributeDerivedFromClass(ITypeSymbol attributeSymbol, ITypeSymbol type, int depth = 10)
+		{
+			attributeSymbol.ThrowOnNull(nameof(attributeSymbol));
+			type.ThrowOnNull(nameof(type));
+			if (depth <= 0 || depth > 100)
+				return false;
+			if (ContainsBaseType(attributeSymbol, type))
+				return true;
+
+			//get symbols from PXContext
```

Comment:
@saloponov  this comment is redundant. It's already obvious from the code what is done here